tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/MsTestIntegration.cs

```diff
@@ -21,6 +22,8 @@ internal static class MsTestIntegration
         internal const IntegrationId IntegrationId = Configuration.IntegrationId.MsTestV2;
         internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MsTestIntegration));
 
+        internal static readonly ThreadLocal<object> IsTestMethodRunnableThreadLocal = new ThreadLocal<object>();
```

Comment:
Haven't looked into the MSTest side to see, but definitely `ThreadLocal<>` and not `AsyncLocal`?

Reply:
Yes, the scenario is a sync method calling other sync method and I need values from the inner method in the EndMethod of the outer one. So a ThreadLocal does the job.